lib/mongo/server/connection_pool/wait_queue.rb

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2014-2019 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  class Server
+    class ConnectionPool
+      class WaitQueue
+        extend Forwardable
+
+        def initialize(address)
+          @address = address
+          @wait_queue = []
+          @mutex = Mutex.new
+        end
+
+        def clear!
```

Comment:
Why is there a bang on this method?

Reply:
I thought it was similar in nature to the `disconnect!` method on `ConnectionPool` and `AvailableStack`, so I put it there. I can take it off if you prefer it without.

Reply:
Changed